flink-yarn/src/main/java/org/apache/flink/yarn/cli/FlinkYarnSessionCli.java

```diff
@@ -500,11 +500,19 @@ protected Configuration applyCommandLineOptionsToConfiguration(CommandLine comma
 		}
 
 		if (commandLine.hasOption(jmMemory.getOpt())) {
-			effectiveConfiguration.setString(JobManagerOptions.JOB_MANAGER_HEAP_MEMORY, commandLine.getOptionValue(jmMemory.getOpt()));
+			String jmMemoryVal = commandLine.getOptionValue(jmMemory.getOpt());
+			if (!jmMemoryVal.toLowerCase().contains("m")) {
```

Comment:
This won't cover other units than "m". How about checking it with `MemoryUnit.hasUnit`?